exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -57,11 +59,24 @@ func (e *kafkaTracesProducer) tracesPusher(_ context.Context, td ptrace.Traces)
 }
 
 func (e *kafkaTracesProducer) Close(context.Context) error {
+	if e.producer == nil {
+		return nil
+	}
 	return e.producer.Close()
 }
 
+func (e *kafkaTracesProducer) start(_ context.Context, _ component.Host) error {
+	producer, err := newSaramaProducer(e.cfg)
```

Comment:
Do you mind just explaining why the producer is created on `Start` instead of `New` ?

Reply:
With component status reporting, the expectation also is that you'd keep trying to connect to Kafka periodically, and report errors otherwise. This would allow an operator to review the issue and use opamp fix the receiver. The way this is managed right now, the component bails out on creation, and the whole collector exits early without recourse.